pkg/querier/querier.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package querier provides tools to interact with the Minder database
+package querier
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/mindersec/minder/internal/db"
+	"github.com/mindersec/minder/pkg/config/server"
+	"github.com/mindersec/minder/pkg/engine/selectors"
+	"github.com/mindersec/minder/pkg/eventer"
+	"github.com/mindersec/minder/pkg/profiles"
+	"github.com/mindersec/minder/pkg/ruletypes"
+)
+
+// Store interface provides functions to execute db queries and transactions
+type Store interface {
+	ProjectHandlers
+	RuleTypeHandlers
+	ProfileHandlers
+	BundleHandlers
+	BeginTx() (Querier, error)
+}
+
+// Querier interface provides functions to interact with the Minder database using transactions
+type Querier interface {
+	ProjectHandlers
+	RuleTypeHandlers
+	ProfileHandlers
+	BundleHandlers
+	CommitTx() error
```

Comment:
I'd probably call this `Commit` and `Cancel`, and document that the object can't be used after calling these.